server/src/models/swing.rs

```diff
@@ -13,20 +13,179 @@
  * limitations under the License.
  */
 
-use std::convert::TryInto;
+//! Implementation of the Swing model type from the [Swing and Slide paper] and
+//! efficient computation of aggregates for models of type Swing as described in
+//! the [ModelarDB paper].
+//!
+//! [Swing and Slide paper]: https://dl.acm.org/doi/10.14778/1687627.1687645
+//! [ModelarDB paper]: https://dl.acm.org/doi/abs/10.14778/3236187.3236215
 
-use datafusion::arrow::array::{Float32Builder, Int32Builder, TimestampMillisecondBuilder};
+use crate::types::{
+    TimeSeriesId, TimeSeriesIdBuilder, Timestamp, TimestampBuilder, Value, ValueBuilder,
+};
 
-/** Public Functions **/
+/// The state the Swing model type needs while fitting a model to a time series
+/// segment.
+struct Swing {
+    /// Maximum relative error for the value of each data point.
+    error_bound: Value,
+    /// Time at which the first value represented by the current model was
+    /// collected.
+    first_timestamp: Timestamp,
+    /// Time at which the last value represented by the current model was
+    /// collected.
+    last_timestamp: Timestamp,
+    /// First value in the segment the current model is fitted to.
+    first_value: f64, // f64 instead of Value to remove casts in fit_value()
```

Comment:
As discussed, maybe find a place to mention why `f64` is used instead of `f32` for values.

Reply:
Extended the module's doc comment as the types are present in the documentation.